### DISCUSSION TITLE
Trouble with correctly import typescript definitions

### DISCUSSION THREAD

**Discussion by ryan-adolphson** [nolimits4web/swiper#7170] [Q&A]
Currently working with Angular 16 (which could be the issue) but I'm not able to import type definitions without getting an error

`Module '"swiper/types"' has no exported member 'SwiperOptions'`

Here is an example on stackblitz if you check the import for `SwiperOptions` the it isn't happy

`import { SwiperOptions } from 'swiper/types';`

Example here => [ https://stackblitz.com/edit/stackblitz-starters-z9ah6k?file=src%2Fapp%2Fapp.component.ts]( https://stackblitz.com/edit/stackblitz-starters-z9ah6k?file=src%2Fapp%2Fapp.component.ts)

I figured it was a module resolution issue but have had no luck going down the rabbit hole... I have to be missing something

**Comment by hajirasyafi** [comment]
try this :
```ts
import {SwiperOptions} from "swiper/types/swiper-options"
```

**Reply by ryan-adolphson** [reply]
Sadly that gives an even worse set of errors where it doesn't like `.d.ts`  and it errors on every single import using `.d.ts`

```
An import path cannot end with a '.d.ts' extension. Consider importing './modules/a11y.js' instead.

import type { A11yOptions } from './modules/a11y.d.ts';
```


I went through my local copy and removed the `.d.ts` and everything works perfectly 

Before
```
import type { A11yOptions } from './modules/a11y.d.ts';
import type { AutoplayOptions } from './modules/autoplay.d.ts';
```

After
```
import type { A11yOptions } from './modules/a11y';
import type { AutoplayOptions } from './modules/autoplay';
```

So I assuming Angular / Webpack are being very picky about the `.d.ts` — I'll fork the repo for now

**Comment by stefanpopi** [comment]
In package.json, if you have "typescript" version < 5, upgrade it to 5+.
That fixed it in my case.

**Reply by Phakeme** [reply]
Thanks, this solution worked for me as well.
